### PR TITLE
lbzip2: fix crash on macOS >= 10.13

### PR DESCRIPTION
**Pull request by philwo**
The current lbzip2 formula is broken since 2018 and crashes on macOS due to it using an incompatible (embedded) version of gnulib.

logological has kindly provided a fork with an updated version of gnulib that works again on recent macOS versions.

Upstream seems unmaintained, thus switching to the fork is probably the most reasonable way forward.

Details are in https://github.com/kjn/lbzip2/issues/20.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Did other package managers adopt this forum as the replacement and/or did upstream? Otherwise we won't accept it either.

**Comment by philwo** [comment]
Upstream seems to have disappeared. Their GitHub account has been inactive for the last years, they don't respond to open issues in lbzip2, no commits, no e-mails to lists anywhere to be found.

I totally understand if you don't want to switch to some other version that promises to contain a fix and nothing malicious and appreciate that you take security seriously. What do you suggest how to move forward with this formula? As it is, it is completely broken - it just crashes on launch.

Unfortunately I can't point to other package managers using the forked version. MacPorts seems to also ship the broken version (https://trac.macports.org/ticket/59290).

What about if I change this PR to instead include a patch that fixes the issue? This should do it: https://lists.gnu.org/archive/html/bug-gnulib/2017-07/txtmumXtpD69v.txt. It applies cleanly and the resulting binary works fine again.

**Comment by jonchang** [comment]
I checked Debian and they still ship 2.5 from an upstream branch (https://github.com/kjn/lbzip2/tree/debian).

@philwo I think we would prefer the patch approach. Would you mind cleaning up and submitting the patch to https://github.com/Homebrew/formula-patches, and update this pull request with the patch? You don't have to wait for the pull request to get merged, you just need to open it and point it to the right commit. We'll merge the two of them together once this this approved.

**Comment by philwo** [comment]
This should be done now. The corresponding PR for the patch is: https://github.com/Homebrew/formula-patches/pull/301

**Review by SeekingMeaning — DISMISSED** [review]

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**
